### PR TITLE
IEnumerable and IQueryable HasValues extension methods

### LINKED ISSUE #51: Extension for inversion of IEnumerable.IsNullOrEmpty
I think it's come up a few times in code reviews but we haven't documented an actual issue to tackle it. There are often times where you'd like to both null check and check for values in a collection, such as conditionally applying a filter.

```cs
public ActionResult Index(IEnumerable<long> ids = null)
{
    Expression<Func<Publication, bool>> filter = (e) => true;

    if (!ids.IsNullOrEmpty())
    {
        filter = filter.AndAlso((e) => ids.Contains(e.Id));
    }

    // ...
}
```

Naming might be up for discussion - we have use `hasValues` and `isEmpty` on our javascript `CollectionUtils`, which are less verbose, but cover the same cases (`isEmpty` returns true if collection is `undefined`, `null`, or `[]`)

I could see `HasValues` to keep it concise, or something like `IsNotNullAndHasValues` to be more explicit with the null check safety as well as keeping consistency with the inverse extension method.

Be sure to add test coverage.

### PR DESCRIPTION
**Pull request by brandongregoryscott**
Resolves #51 Extension for inversion of IEnumerable.IsNullOrEmpty 

- [x] Related GitHub issue(s) linked in PR description
- [x] Destination branch merged, built and tested with your changes
- [x] Code formatted and follows best practices and patterns
- [x] Code builds cleanly (no _additional_ warnings or errors)
- [x] Manually tested
- [x] Automated tests are passing
- [x] No _decreases_ in automated test coverage
- [x] Documentation updated (readme, docs, comments, etc.)
- [-] Localization: No hard-coded error messages in code files (_minimally_ in string constants)

### CODE REVIEW

**Review by SaidShah — APPROVED** [review]

**Review by jstevenson — APPROVED** [review]
Thanks for removing unnecessary comments 👍

File: src/AndcultureCode.CSharp.Extensions/IQueryableExtensions.cs [hunk 1]
```diff
@@ -10,39 +10,43 @@ namespace AndcultureCode.CSharp.Extensions
     /// </summary>
     public static class IQueryableExtensions
     {
+
```
Comment (wintondeshong) [inline src/AndcultureCode.CSharp.Extensions/IQueryableExtensions.cs:13]: Nitpick: extra newline

File: test/AndcultureCode.CSharp.Extensions.Tests/IEnumerableExtensionsTest.cs [hunk 2]
```diff
@@ -28,15 +140,14 @@ public void IsEmpty_Given_NoArguments_When_NotEmpty_Returns_False()
...
         #region IsEmpty (Given Predicate)
 
         [Fact]
         public void IsEmpty_Given_Predicate_When_Empty_Returns_True()
         {
-            new List<string>()       // Arrange
-                .IsEmpty()           // Act
-                    .ShouldBeTrue(); // Assert
+            new List<string>()                    // Arrange
+                .IsEmpty((e) => e == "something") // Act
```
Comment (wintondeshong) [inline test/AndcultureCode.CSharp.Extensions.Tests/IEnumerableExtensionsTest.cs:149]: Minor; but perhaps randomize `something`

**Review by wintondeshong — APPROVED** [review]
uber minor stuff - looks good @brandongregoryscott

**Comment by wintondeshong** [comment]
@all-contributors please add @brandongregoryscott for code, tests, review and documentation

**Comment by wintondeshong** [comment]
@all-contributors please add @wintondeshong for code, tests, review and documentation

**Pull request merged**